great_expectations/datasource/fluent/redshift_datasource.py

```diff
@@ -57,32 +55,34 @@ class RedshiftDatasource(SQLDatasource):
         name: The name of this Redshift datasource.
         connection_string: The SQLAlchemy connection string used to connect to the Redshift
             database. This will use a redshift with psycopg2. For example:
-            "redshift+psycopg2://username@host.amazonaws.com:5439/database"
-        connection_details: A RedshiftConnectionDetails object defining the connection details.
+            "redshift+psycopg2://user:password@host.amazonaws.com:5439/database?sslmode=sslmode".
+            Alternatively, a RedshiftConnectionDetails object can be used.
         If connection_details is used, connection_string cannot also be provided.
         assets: An optional dictionary whose keys are TableAsset or QueryAsset names and whose
             values are TableAsset or QueryAsset objects.
     """
 
     type: Literal["redshift"] = "redshift"  # type: ignore[assignment] # This is a hardcoded constant
-    connection_string: Union[ConfigStr, RedshiftDsn]
+    connection_string: Union[RedshiftConnectionDetails, ConfigStr, RedshiftDsn]  # type: ignore[assignment] # Deviation from parent class as individual args are supported for connection
```

Comment:
I think you can remove the `RedshiftConnectionDetails` since you always coerce to a `str`.
Pros: Someone who goes `redshift_datasource.connection_string` doesn't have to handle the different union types.
Cons: Autocomplete will only give the string argument. One will have to do type ignores when passing in these non-str types.
We don't annotate the `dict` argument.

I've gone back and forth. I think we can leave this with the type union. I'd add the dict type, probably as a TypedDict with the right keys.